ember/app/ui/components/work-item-list/component.js

```diff
@@ -3,7 +3,10 @@ import Component from "@glimmer/component";
 
 export default class WorkItemListComponent extends Component {
   @action
-  fetchMore() {
-    this.args.query.fetchMore();
+  async fetchMore() {
+    await this.args.query.fetchMore();
+    if (this.args.fetchMoreSupplement) {
+      await this.args.fetchMoreSupplement.perform(this.args.query.value);
```

Comment:
This implies that the component knows exactly what is passed as the `fetchMoreSupplement` argument. If you did a type check here you could either call `.perform` if its a task or simply call the argument if it is a regular function or do nothing if its undefined.

Reply:
Changed to generic function call